version: '3.7'

services:
  app:
    build:
      context: .
    ports:
      - 8070:8070

    environment:
      - DATABASE_URL=postgres://postgres:postgres@postgres:5432/moove_one
      - DB_HOST=postgres
      - DB_USER=postgres
      - DB_NAME=moove_one
      - DB_PASSWORD=postgres
    volumes:
      - ./:/app
    command: >
      sh -c "python3 manage.py migrate &&
             python3 manage.py run_data_exceptions &&
             python3 manage.py run_data_trips &&
             python3 manage.py run_data_vehicles &&
             python3 manage.py clear_all_tables &&
             python3 manage.py runserver 0.0.0.0:8070"
    networks:
      - app-network
    depends_on:
      - postgres
      - redis

  postgres:
    image: postgres:10-alpine
    environment:
      - POSTGRES_DB=moove_one
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    volumes:
      - postgres:/data/postgres
    networks:
      - app-network
    restart: unless-stopped

  redis:
    image: redis:alpine
  celery:
    build:
      context: ./
      dockerfile: Dockerfile
    restart: unless-stopped
    command: celery -A app worker -l info
    volumes:
      - ./:/app
    networks:
      - app-network
    env_file:
      - ./.env
    depends_on:
      - app
      - redis

networks:
  app-network:
    driver: bridge

volumes:
  postgres:
  celery:
